compiler/rustc_session/src/options.rs

```diff
@@ -2062,6 +2208,26 @@ pub mod parse {
 
         true
     }
+
+    pub(crate) fn parse_allow_partial_mitigations(
+        slot: &mut Vec<MitigationEnablement>,
+        v: Option<&str>,
+    ) -> bool {
+        match v {
+            Some(s) => {
+                for sub in s.split(',') {
+                    let (sub, enabled) =
+                        if sub.starts_with('!') { (&sub[1..], false) } else { (sub, true) };
```

Comment:
Using `!` for negation on the command-line is non-standard. It might also cause shell command parsing issues, because `!` is a history search and insert metacharacter in some common shells.

The standard CLI negation prefix is `no-`, I suggest we use that instead.

The dev guide says to avoid `no-` for boolean flags, but I don't think that applies to more complex options:
https://github.com/rust-lang/rust/blob/1be6b13be73dc12e98e51b403add4c41a0b77759/src/doc/rustc-dev-guide/src/cli.md?plain=1#L15-L16

Reply:
I basically put `!` as a placeholder. Let's wait with it for a little bit of time until we figure out what to do.